packages/codec/lib/types/decoding.ts

```diff
@@ -2,67 +2,236 @@ import { DecoderContext } from "./contexts";
 import * as AbiTypes from "./abi";
 import { Types, Values } from "../format";
 
+/**
+ * A type representing a transaction (calldata) decoding.  As you can see, these come in four types,
+ * each of which is documented separately.
+ */
 export type CalldataDecoding = FunctionDecoding | ConstructorDecoding | MessageDecoding | UnknownDecoding;
+
+/**
+ * A type representing a log (event) decoding.  As you can see, these come in two types, each of which
+ * is documented separately.
+ */
 export type LogDecoding = EventDecoding | AnonymousDecoding;
 
+/**
+ * This is a type for recording what decoding mode a given decoding was produced in.  There are two
+ * decoding modes, full mode and ABI mode.  In ABI mode, decoding is done purely based on the ABI JSON.
+ * Full mode, by contrast, additionally uses AST information to produce a more informative decoding.
+ * For more on full mode and ABI mode, see the README.
```

Comment:
(link)

Reply:
actually, how do you feel about making this the canonical home for information about decoding modes, and have the README, etc. link here?

Reply:
Ehhhh.  I feel like that's important enough to go in the README?  If you're going to use the decoder you should know about this.  I don't want such an important thing being buried here.

Reply:
(capturing as note for follow-on issue)